week3/homework/exercises/1-joke-api/script.js

```diff
@@ -1,18 +1,33 @@
 /**
  * 1. Chuck Norris programs do not accept input
- * 
+ *
  * `GET` a random joke inside the function, using the API: http://www.icndb.com/api/
- * (use `node-fetch`) and print it to the console. 
+ * (use `node-fetch`) and print it to the console.
  * Make use of `async/await` and `try/catch`
- * 
+ *
  * Hints
  * - To install node dependencies you should first initialize npm
  * - Print the entire response to the console to see how it is structured.
  */
+const fetch = require('node-fetch');
 
-function printChuckNorrisJoke() {
+// fetch('http://api.icndb.com/jokes/random')
+//   .then( res => res.json())
+//   .then(json => console.log(json));
+
+
+
+async function printChuckNorrisJoke() {
   // YOUR CODE GOES IN HERE
+  try {
+    const response = await fetch('http://api.icndb.com/jokes/random');
+    const data = await response.json()
+   return data
+  } catch(err) {
+    console.log(err)
+  }
 
 }
 
-printChuckNorrisJoke();
\ No newline at end of file
+printChuckNorrisJoke().then(data => console.log(data));
```

Comment:
I won't do it this way. It is a bit difficult to read. Also, as far as I see, the return value `data` is not a promise. Does this code work?

In this case you don't really need a return. As the name of the function suggests, `print` is in it so just console inside the function